docs/extending_wsimod.md

```diff
@@ -0,0 +1,67 @@
+# Extending and customising WSIMOD functionality
+
+WSIMOD has many features built-in, but undoubtedly, to model real life scenarios, you will
+need to customise it to your specific needs.
+
+WSIMOD offers 3 ways of doing this customisation of increased complexity - and flexibility:
+overrides, patches and custom classes.
+
+## Overrides
+
+[**NOTE**: I have not seen this in action in any example or demo, and `apply_overrides` does not seem to be used ever, so it is unclear how this is actually used.]
```

Comment:
Is there enough material now in the tests to finish this bit

Reply:
Yes, there is. I'll work on this tomorrow morning, hoping to finish my bit, at least.